go-controller/pkg/ovn/namespace.go

```diff
@@ -105,28 +105,53 @@ func (oc *Controller) multicastUpdateNamespace(ns *kapi.Namespace, nsInfo *names
 	}
 
 	var err error
+	nsInfo.multicastEnabled = enabled
 	if enabled {
 		err = oc.createMulticastAllowPolicy(ns.Name, nsInfo)
 	} else {
-		err = deleteMulticastAllowPolicy(ns.Name)
+		err = deleteMulticastAllowPolicy(ns.Name, nsInfo)
 	}
 	if err != nil {
 		klog.Errorf(err.Error())
 		return
 	}
-
-	nsInfo.multicastEnabled = enabled
 }
 
 // Cleans up the multicast policy for this namespace if multicast was
 // previously allowed.
 func (oc *Controller) multicastDeleteNamespace(ns *kapi.Namespace, nsInfo *namespaceInfo) {
 	if nsInfo.multicastEnabled {
-		if err := deleteMulticastAllowPolicy(ns.Name); err != nil {
+		nsInfo.multicastEnabled = false
+		if err := deleteMulticastAllowPolicy(ns.Name, nsInfo); err != nil {
 			klog.Errorf(err.Error())
 		}
 	}
-	nsInfo.multicastEnabled = false
+}
+
+// returns the UUID of a port_group for the given namespace if it does not exist create it
```

Comment:
comment could use some punctuation. (like maybe a period after "namespace". And appropriate capital letters)

Reply:
fixed